flink-runtime/src/main/java/org/apache/flink/runtime/metrics/dump/MetricQueryService.java

```diff
@@ -119,6 +123,36 @@ public void onReceive(Object message) {
 		}
 	}
 
+	public void unregister(Metric metric) {
+		if (metric instanceof Counter) {
+			this.counters.remove(metric);
+		} else if (metric instanceof Gauge) {
+			this.gauges.remove(metric);
+		} else if (metric instanceof Histogram) {
+			this.histograms.remove(metric);
+		} else if (metric instanceof Meter) {
+			this.meters.remove(metric);
+		}
+	}
+
+	private void unregisterOversizedMetrics(MetricDumpSerialization.MetricSerializationResult serializationResult) {
+		for (Metric metric : serializationResult.counters.keySet()) {
```

Comment:
this line is incredibly misleading; based on the naming one would assume that all metrics are unregistered.